server/main.lua

```diff
@@ -59,36 +59,64 @@ end)
 ----   Functions   ----
 -----------------------
 
+--- Gives the user the keys to the vehicle
+--- @param source number ID of the player
+--- @param plate string The plate number of the vehicle.
 function GiveKeys(source, plate)
+    local citizenid = getCitizenId(source)
+
+    if not citizenid then return end
```

Comment:
In what scenario would citizen id be nil?

If this is an "impossible" condition that would violate core assumptions about how core works, then either remove this line altogether, or assert that citizenid is not nil.

return end should only be used in cases where we want to silently fail.

Reply:
exports.qbx_core:GetPlayer(source) would have to return nil. Do you expect such a situation?
https://github.com/Qbox-project/qbx_vehiclekeys/blob/aae0c26ff4cb46bc372846f675c70b97b4356730/server/main.lua#L44-L46

Reply:
Only if source passed into it was invalid. So this should be an assertion or unhandled case me thinks. Ok to leave for now.